modules/cnf-topology-aware-lifecycle-operator-precache-feature.adoc

```diff
@@ -0,0 +1,234 @@
+
+. When you want to start the update, apply the `ClusterGroupUpgrade` CR by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f clustergroupupgrades-group-du.yaml
+----
+
+.Verification
+
+. Check if the `ClusterGroupUpgrade` CR exists in the Hub cluster by running the following command:
++
+[source,terminal]
+----
+$ oc get cgu -A
+----
++
+.Example output
++
+[source,terminal]
+----
+NAMESPACE          NAME              AGE
+ztp-group-du-sno   du-upgrade-4918   10s <1>
+----
+<1> The CR is created.
+
+. Check the status of the pre-caching task by running the following command:
++
+[source,terminal]
+----
+$ oc get cgu -n ztp-group-du-sno du-upgrade-4918 -o jsonpath='{.status}' | jq
+----
++
+.Example output
++
+[source,json]
+----
+{
+  "conditions": [
+    {
+      "lastTransitionTime": "2022-01-27T19:07:24Z",
+      "message": "Precaching is not completed (required)", <1>
+      "reason": "PrecachingRequired",
+      "status": "False",
+      "type": "Ready"
+    },
+    {
+      "lastTransitionTime": "2022-01-27T19:07:24Z",
+      "message": "Precaching is required and not done",
+      "reason": "PrecachingNotDone",
+      "status": "False",
+      "type": "PrecachingDone"
+    },
+    {
+      "lastTransitionTime": "2022-01-27T19:07:34Z",
+      "message": "Pre-caching spec is valid and consistent",
+      "reason": "PrecacheSpecIsWellFormed",
+      "status": "True",
+      "type": "PrecacheSpecValid"
+    }
+  ],
+  "precaching": {
+    "clusters": [
+      "cnfdb1" <2>
+    ],
+    "spec": {
+      "platformImage": "image.example.io"
+ },
+    "status": {
+      "cnfdb1": "Active"
+----
+<1> Displays that the update is in progress.
+<2> Displays the list of identified clusters.
+
+. Check the status of the pre-caching job by running the following command:
++
+[source,terminal]
+----
+$ oc get jobs,pods -n openshift-talo-pre-cache
+----
++
+.Example output
++
+[source,terminal]
+----
+NAME                  COMPLETIONS   DURATION   AGE
+job.batch/pre-cache   0/1           3m10s      3m10s
+
+NAME                     READY   STATUS    RESTARTS   AGE
+pod/pre-cache--1-9bmlr   1/1     Running   0          3m10s
+----
+
+ . Check the status of the `ClusterGroupUpgrade` CR by running the following command:
++
+[source,terminal]
+----
+$ oc get cgu -n ztp-group-du-sno du-upgrade-4918 -o jsonpath='{.status}' | jq
+----
++
+.Example output
++
+[source,json]
+----
+conditions": [
+    {
+      "lastTransitionTime": "2022-01-27T19:30:41Z",
+      "message": "The ClusterGroupUpgrade CR has all clusters compliant with all the managed policies",
+      "reason": "UpgradeCompleted",
+      "status": "True",
+      "type": "Ready"
+    },
+    {
+      "lastTransitionTime": "2022-01-27T19:28:57Z",
+      "message": "Precaching is completed",
+      "reason": "PrecachingCompleted",
+      "status": "True",
+      "type": "PrecachingDone" <1>
+    }
+----
+<1> The pre-cache tasks are done.
```

Comment:
We need to tell user how to start upgrade from this point on. Basically, user needs to know that they have to delete and reapply the same CGU to start upgrade after pre-caching is completed.